### PR TITLE
Cache plot counts

### PR DESCRIPTION
**Pull request by RickMohr**
Connects #2274

### CODE REVIEW

File: opentreemap/treemap/ecocache.py [hunk 1]
```diff
@@ -7,32 +7,42 @@
...
+    return _get_or_compute(prefix, version, filter, compute_value)
 
-def cache_benefits(instance, filter, benefits):
-    if settings.USE_ECO_CACHE:
-        key = _get_key(instance, filter)
-        cache.set(key, benefits, TIMEOUT)
+
+def _get_or_compute(prefix, version, filter, compute_value):
+    if not settings.USE_ECO_CACHE:
+        return None
```
Comment (jwalgran) [inline opentreemap/treemap/ecocache.py]: It looks like we need to `return compute_value()` here rather than `return None`.

**Comment by jwalgran** [comment]
I was able to repeat the test failures reported by CI

```
======================================================================
ERROR: test_benefit_search (treemap.tests.test_urls.TreemapUrlTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/otm/app/opentreemap/treemap/tests/test_urls.py", line 242, in test_benefit_search
    'treemap/partials/eco_benefits.html')
  File "/usr/local/otm/app/opentreemap/treemap/tests/test_urls.py", line 41, in assert_template
    response = self.assert_status_code(url, 200)
  File "/usr/local/otm/app/opentreemap/treemap/tests/test_urls.py", line 29, in assert_status_code
    response = send(url, data)
  File "/usr/local/otm/env/local/lib/python2.7/site-packages/django/test/client.py", line 500, in get
    **extra)
  File "/usr/local/otm/env/local/lib/python2.7/site-packages/django/test/client.py", line 303, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/usr/local/otm/env/local/lib/python2.7/site-packages/django/test/client.py", line 379, in generic
    return self.request(**r)
  File "/usr/local/otm/env/local/lib/python2.7/site-packages/django/test/client.py", line 466, in request
    six.reraise(*exc_info)
  File "/usr/local/otm/env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/otm/app/opentreemap/treemap/decorators.py", line 45, in wrapper
    return view_fn(request, instance, *args, **kwargs)
  File "/usr/local/otm/env/local/lib/python2.7/site-packages/django/views/decorators/http.py", line 158, in inner
    response = func(request, *args, **kwargs)
  File "/usr/local/otm/env/local/lib/python2.7/site-packages/django_tinsel/decorators.py", line 79, in wrapper
    params = callable_or_dict(request, *args, **wrapper_kwargs)
  File "/usr/local/otm/app/opentreemap/treemap/views/tree.py", line 74, in search_tree_benefits
    benefits, basis = get_benefits_for_filter(filter)
  File "/usr/local/otm/app/opentreemap/treemap/ecobenefits.py", line 417, in get_benefits_for_filter
    ft_benefit_groups, ft_basis = _benefits_for_class(C, filter)
TypeError: 'NoneType' object is not iterable
```

```
======================================================================
ERROR: treemap.tests.test_ecobenefits (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: treemap.tests.test_ecobenefits
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/loader.py", line 252, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib/python2.7/unittest/loader.py", line 230, in _get_module_from_name
    __import__(name)
  File "/usr/local/otm/app/opentreemap/treemap/tests/test_ecobenefits.py", line 22, in <module>
    from treemap.ecocache import cache_benefits, get_cached_benefits
ImportError: cannot import name cache_benefits
```

**Comment by jwalgran** [comment]
I ran tests with a instance with 260,000+ trees. And saw both the eco and plot count cached in the database.

```
$ time curl -I http://localhost:6060/instance/benefit/search
HTTP/1.1 200 OK
...
real	0m3.465s

$ time curl -I http://localhost:6060/edmonton/benefit/search
HTTP/1.1 200 OK
...
real	0m0.318s

```

```
otm=# select cache_key,expires from cache;
                            cache_key                            |        expires
-----------------------------------------------------------------+------------------------
 :1:count/plots/instance/267547/6666cd76f96956469e7be39d750cc7d9 | 2015-10-29 06:02:56+00
 :1:eco/trees/instance/1/6666cd76f96956469e7be39d750cc7d9        | 2015-10-29 06:02:57+00
```

**Comment by RickMohr** [comment]
@jwalgran, sorry about that. Fixed up now.

**Comment by jwalgran** [comment]
Tests are green. Lint is gone. I was able to see the count and eco cached in the database and setting `USE_ECO_CACHE = False` turned off the caching but still produced results.

:+1:

**Pull request merged**